src/subdomains/core/liquidity-management/services/liquidity-management.service.ts

```diff
@@ -38,8 +40,42 @@ export class LiquidityManagementService {
     }
   }
 
+  //*** PUBLIC API ***//
+
+  async buyLiquidity(assetId: number, amount: number): Promise<PipelineId> {
+    const rule = await this.findRuleByAssetOrThrow(assetId);
+
+    if (!rule.deficitStartAction) {
+      throw new BadRequestException(`Rule ${rule.id} does not support liquidity deficit path`);
+    }
+
+    const liquidityState = { deficit: amount, redundancy: 0 };
+
+    return this.executeRule(rule, liquidityState);
+  }
+
+  async sellLiquidity(assetId: number, amount: number): Promise<PipelineId> {
+    const rule = await this.findRuleByAssetOrThrow(assetId);
+
+    if (!rule.redundancyStartAction) {
+      throw new BadRequestException(`Rule ${rule.id} does not support liquidity redundancy path`);
+    }
+
+    const liquidityState = { deficit: 0, redundancy: amount };
+
+    return this.executeRule(rule, liquidityState);
+  }
```

Comment:
Should we check for current liquidity state and combine both inputs?

Reply:
lets keep it like that